packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -332,9 +342,11 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
             this.resolveNewValue(updatedValue, newSuggestions);
           } else {
             this.suggestionStore.updateSuggestions(newSuggestions);
-            this.setState({
-              suggestionsLoading: false
-            });
+            if (!this._isUnmounted) {
```

Comment:
Another option is to set `this.currentPromise` to `undefined` in `unmount`. It does not need to introduce the `isUnmount` field.

Reply:
Good suggestion! I would like to try it.